apm/otel_monitor.go

```diff
@@ -451,13 +450,14 @@ func stripDocument(doc bson.Raw) (bson.Raw, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "enumerating document elements")
 	}
-	strippedDocument := bson.D{}
-	for _, elem := range elems {
+	strippedDocument := make(bson.D, len(elems))
+	for i, elem := range elems {
+		elem := elem
 		elemValue, err := stripValue(elem.Value())
 		if err != nil {
 			return nil, errors.Wrap(err, "stripping document values")
 		}
-		strippedDocument = append(strippedDocument, bson.E{Key: elem.Key(), Value: elemValue})
+		strippedDocument[i] = bson.E{Key: elem.Key(), Value: elemValue}
```

Comment:
It's slightly better to declare the array size and then do this, I can revert if it makes it more complex

Reply:
That seems fine to me, it's logically the same.